bench/src/bulk.rs

```diff
@@ -51,27 +62,72 @@ async fn server(mut incoming: quinn::Incoming) -> Result<()> {
     let quinn::NewConnection {
         mut uni_streams, ..
     } = handshake.await.context("handshake failed")?;
-    let mut stream = uni_streams
-        .next()
-        .await
-        .ok_or_else(|| anyhow!("accepting stream failed"))??;
-    trace!("stream established");
+
+    loop {
+        let mut stream = match uni_streams.next().await {
+            None => return Ok(()),
+            Some(Err(quinn::ConnectionError::ApplicationClosed(_))) => return Ok(()),
+            Some(Err(e)) => return Err(e).context("accepting stream failed"),
+            Some(Ok(stream)) => stream,
+        };
+        trace!("stream established");
+
+        let _jh: tokio::task::JoinHandle<Result<()>> = tokio::spawn(async move {
+            let mut n = 0u64;
+            while let Some((data, offset)) = stream.read_unordered().await? {
+                n = n.max(offset + data.len() as u64);
+            }
+
+            Ok(())
+        });
+    }
+}
+
+#[derive(Debug)]
```

Comment:
Since `SendResult` is used by `send_data_on_stream()`, let's move the type below the function defn.

Reply:
That's the reason why I have it above: I like to learn about dependencies before the places where they are used. Allows to read from top to bottom :-) 

But I will change it if preferred

Reply:
I tend to come down on @djc's side here as well, as it's easy to overflow your working memory with dependencies before you get to the root, and the root can be a useful guide to understand the significance of the dependencies.